mlflow/artifacts/__init__.py

```diff
@@ -104,7 +106,34 @@ def list_artifacts(
     run_id: Optional[str] = None,
     artifact_path: Optional[str] = None,
     tracking_uri: Optional[str] = None,
-):
+) -> list[FileInfo]:
+    run_artifacts = _list_run_artifacts(
+        artifact_uri=artifact_uri,
+        run_id=run_id,
+        artifact_path=artifact_path,
+        tracking_uri=tracking_uri,
+    )
+    if (
+        run_artifacts
+        or (artifact_uri and not artifact_uri.startswith("runs:/"))
+        or (artifact_path and "/" in artifact_path)
```

Comment:
Logged model names can't include `/`.

Reply:
Can we add an inline comment in the code explaining this?